tests/zfs-tests/tests/functional/dbufs/dbufstats_001_pos.ksh

```diff
@@ -0,0 +1,87 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2017, Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/tests/functional/dbufs/dbufs.kshlib
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/include/math.shlib
+
+#
+# DESCRIPTION:
+# Ensure stats presented in /proc/spl/kstat/zfs/dbuf_stats are correct
+# based on /proc/spl/kstat/zfs/dbufs.
+#
+# STRATEGY:
+# 1. Generate a file with random data in it
+# 2. Store output from dbufs kstat
+# 3. Store output from dbuf_stats kstat
+# 4. Compare stats presented in dbuf_stats with stat generated using
+#    dbufstat.py and the dbufs kstat output
+#
+
+DBUFSTATS_FILE=$(mktemp $TEST_BASE_DIR/dbufstats.out.XXXXXX)
+DBUFS_FILE=$(mktemp $TEST_BASE_DIR/dbufs.out.XXXXXX)
+
+function cleanup
+{
+	log_must rm -f $TESTDIR/file $DBUFS_FILE $DBUFSTATS_FILE
+}
+
+function testdbufstat # stat_name dbufstat_filter
+{
+        name=$1
+        filter=""
+
+        [[ -n "$2" ]] && filter="-F $2"
+
+        verify_eq \
+	    $(grep -w "$name" "$DBUFSTATS_FILE" | awk '{ print $3 }') \
+	    $(dbufstat.py -bxn -i "$DBUFS_FILE" "$filter" | wc -l) \
+	    "$name"
+}
+
+verify_runnable "both"
+
+log_assert "dbufstats produces correct statistics"
+
+log_onexit cleanup
+
+log_must file_write -o create -f "$TESTDIR/file" -b 1048576 -c 20 -d R
+log_must zpool sync
+
+sleep 5
```

Comment:
Sleeping shouldn't be required, why was this needed?

Reply:
It's a remnant from before `zpool sync` was called. I'll remove it.